app/src/org/commcare/CommCareApplication.java

```diff
@@ -947,7 +957,16 @@ public void onServiceConnected(ComponentName className, IBinder service) {
                         }
                     }
 
-                    XPathErrorLogger.registerStorage(getUserStorage(XPathErrorEntry.STORAGE_KEY, XPathErrorEntry.class));
+                    // Switch the regular and forece close loggers over to using user storage while
+                    // there is a session
+                    Logger.registerLogger(new AndroidLogger(getUserStorage(AndroidLogEntry.STORAGE_KEY,
+                            AndroidLogEntry.class)));
+                    ForceCloseLogger.registerStorage(getUserStorage(ForceCloseLogEntry.STORAGE_KEY,
+                            ForceCloseLogEntry.class));

```

Comment:
These logger registrations could be lifted into a helper like `void setupLoggers(boolean isUserStorageAvailable)`